Resource.cs

```diff
@@ -0,0 +1,193 @@
+
+        [JsonPropertyName("device")]
+        public string? Device { get; set; }
+
+        [JsonPropertyName("clientIdentifier")]
+        public string? ClientIdentifier { get; set; }
+
+        [JsonPropertyName("createdAt")]
+        public DateTime CreatedAt { get; set; }
+
+        [JsonPropertyName("lastSeenAt")]
+        public DateTime LastSeenAt { get; set; }
+
+        [JsonPropertyName("provides")]
+        public string? Provides { get; set; }
+
+        [JsonPropertyName("ownerId")]
+        public int? OwnerId { get; set; }
+
+        [JsonPropertyName("sourceTitle")]
+        public string? SourceTitle { get; set; }
+
+        [JsonPropertyName("publicAddress")]
+        public string? PublicAddress { get; set; }
+
+        [JsonPropertyName("accessToken")]
+        public string? AccessToken { get; set; }
+
+        [JsonPropertyName("owned")]
+        public bool Owned { get; set; }
+
+        [JsonPropertyName("home")]
+        public bool Home { get; set; }
+
+        [JsonPropertyName("synced")]
+        public bool Synced { get; set; }
+
+        [JsonPropertyName("relay")]
+        public bool Relay { get; set; }
+
+        [JsonPropertyName("presence")]
+        public bool Presence { get; set; }
+
+        [JsonPropertyName("httpsRequired")]
+        public bool HttpsRequired { get; set; }
+
+        [JsonPropertyName("publicAddressMatches")]
+        public bool PublicAddressMatches { get; set; }
+
+        [JsonPropertyName("dnsRebindingProtection")]
+        public bool DnsRebindingProtection { get; set; }
+
+        [JsonPropertyName("natLoopbackSupported")]
+        public bool NatLoopbackSupported { get; set; }
+
+        [JsonPropertyName("connections")]
+        public IList<Connection>? Connections { get; set; }
+        public Uri? LocalUri { get; set; }
+        public Uri? Uri { get; set; }
+
+        public static async Task<Resource[]?> GetAccountResources() {
+            try {
+                HttpRequestMessage requestMessage = new(HttpMethod.Get, $"https://plex.tv/api/v2/resources?includeHttps=1&includeIPv6=1&X-Plex-Token={App.Token}&X-Plex-Client-Identifier=PlexampRPC");
+                requestMessage.Headers.Add("Accept", "application/json");
+
+                HttpResponseMessage sendResponse = await MainWindow.httpClient.SendAsync(requestMessage);
+                sendResponse.EnsureSuccessStatusCode();
+
+                JsonDocument responseJson = JsonDocument.Parse(await sendResponse.Content.ReadAsStringAsync());
+                Resource[]? source_resources = JsonSerializer.Deserialize<Resource[]>(responseJson.RootElement);
+                List<Resource>? filtered_resources = new();
+                if (source_resources == null || source_resources.Length == 0) {
+                    Console.WriteLine("WARN: No servers found");
+                    return null;
+                }
+
+                for (int i = 0; i < source_resources.Length; i++) {
```

Comment:
Is there a reason a for loop is used here instead of foreach /gen

for loops are a lot harder for me to visualize in my head so I try to use foreach whenever i can

Reply:
At the time I thought I wanted to modify the original resources list, but I ended up making a separate filtered list. Could be changed to foreach with no problem